src/magick-image.ts

```diff
@@ -552,6 +553,24 @@ export class MagickImage extends NativeInstance implements IMagickImage {
 
     get width(): number { return ImageMagick._api._MagickImage_Width_Get(this._instance); }
 
+    addProfile(name:string, data:Uint8Array) {
+        Exception.use(exception => {
+            _withString(name, namePtr => {
+                const length = data.byteLength;
+                let dataPtr = 0;
+                try {
```

Comment:
This block seems to be duplicated a couple times now. We should probably create a `_withArray` method that wraps this logic. Can be done after this PR gets merged.

Reply:
I will do it after this PR.